homeassistant/components/dyson/fan.py

```diff
@@ -171,22 +200,69 @@ def service_handle(service):
             DYSON_DOMAIN, SERVICE_SET_TIMER, service_handle, schema=SET_TIMER_SCHEMA
         )
 
-        hass.services.register(
-            DYSON_DOMAIN,
-            SERVICE_SET_DYSON_SPEED,
-            service_handle,
-            schema=SET_DYSON_SPEED_SCHEMA,
-        )
-
 
 class DysonFanEntity(DysonEntity, FanEntity):
     """Representation of a Dyson fan."""
 
+    @property
+    def speed(self):
+        """Return the current speed."""
+        return SPEED_DYSON_TO_HA[self._device.state.speed]
+
+    @property
+    def speed_list(self) -> list:
+        """Get the list of available speeds."""
+        return [SPEED_LOW, SPEED_MEDIUM, SPEED_HIGH]
```

Comment:
We can break out the list to a module level constant that we return here, since the list is constant.

Reply:
There's no speed off?

Reply:
I simply copied the `speed_list` property from `DysonPureCoolEntity` here. Should I add speed off?

Reply:
Only if it's supported by the device. It's ok to keep it like this. I just noted that it wasn't in there.